### PR TITLE
Add service to update currency in a recurrent way

### PR DESCRIPTION
**Pull request by tgallice**
### What does this PR do?

Add a tool to update the currency

Fixes https://github.com/upfluence/cs/issues/570

### Good PR checklist

- [ ] Title makes sense
- [ ] Is against the correct branch
- [ ] Only addresses one issue
- [ ] Properly assigned
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Properly labeled

### CODE REVIEW

File: lib/upfluence.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+require 'upfluence/bank/currency_updater.rb'
```
Comment (AlexisMontagne) [inline lib/upfluence.rb]:
```suggestion
require 'upfluence/bank/currency_updater'
```

File: lib/upfluence/bank/currency_updater.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module Upfluence
+  module Bank
+    class CurrencyUpdater
```
Comment (AlexisMontagne) [inline lib/upfluence/bank/currency_updater.rb]: if you do something that generic (that as nothing to do with Currency Update) maybe keep the name  generic as well

Comment (tgallice) [reply lib/upfluence/bank/currency_updater.rb]: I have hesitated, you're right :+1:

File: lib/upfluence/utils/interval_updater.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module Upfluence
+  class IntervalUpdater
```
Comment (AlexisMontagne) [inline lib/upfluence/utils/interval_updater.rb]: `IntervalExecutor`? Updater might be a little too opinionated

File: lib/upfluence/utils.rb [hunk 4]
```diff
@@ -1,3 +1,4 @@
+require 'upfluence/utils/interval_updater'
```
Comment (BaptisteViry) [inline lib/upfluence/utils.rb]:
```suggestion
require 'upfluence/utils/interval_executor'
```

File: lib/upfluence/utils/interval_executor.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+module Upfluence
+  class IntervalExecutor
```
Comment (BaptisteViry) [inline lib/upfluence/utils/interval_executor.rb]: nest this class in a `module Utils` no ? to be consistent with the file path

Comment (tgallice) [reply lib/upfluence/utils/interval_executor.rb]: arf lol thk

**Review by BaptisteViry — APPROVED** [review]

**Pull request merged**